spring-cloud-kubernetes-client-discovery/src/test/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesInformerDiscoveryClientAutoConfigurationApplicationContextTests.java

```diff
@@ -62,11 +65,30 @@ static void afterAll() {
 	void discoveryEnabledDefault() {
 		setup("spring.main.cloud-platform=KUBERNETES", "spring.cloud.config.enabled=false");
 		applicationContextRunner.run(context -> {
+			assertThat(context).hasSingleBean(KubernetesInformerDiscoveryClient.class);
+			assertThat(context).doesNotHaveBean(KubernetesInformerReactiveDiscoveryClient.class);
+
 			assertThat(context).hasSingleBean(KubernetesDiscoveryClientHealthIndicatorInitializer.class);
+			assertThat(context).doesNotHaveBean(ReactiveDiscoveryClientHealthIndicator.class);
+
+			assertNonSelectiveNamespacesBeansPresent(context);
+			assertSelectiveNamespacesBeansMissing(context);
+		});
+	}
+
+	@Test
+	void discoveryEnabledDefaultWithSelectiveNamespaces() {
+		setup("spring.main.cloud-platform=KUBERNETES", "spring.cloud.config.enabled=false",
+				"spring.cloud.kubernetes.discovery.namespaces=a,b,c");
+		applicationContextRunner.run(context -> {
 			assertThat(context).hasSingleBean(KubernetesInformerDiscoveryClient.class);
-			assertThat(context).hasSingleBean(SharedInformerFactory.class);
```

Comment:
assertions like : 

```
assertThat(context).hasSingleBean(SharedInformerFactory.class);
assertThat(context).getBeans(SharedIndexInformer.class).hasSize(2);
assertThat(context).getBeans(Lister.class).hasSize(2);
```

have not been removed, they moved to `assertNonSelectiveNamespacesBeansMissing` and `assertSelectiveNamespacesBeansPresent` methods. I decided to move them and make them slightly more generic because they are re-used in blocking and reactive tests